web/src/installers/index.ts

```diff
@@ -28,20 +28,20 @@ export interface KubernetesConfig {
   certKey?: string;
 }
 
-const kubernetesConfigSchema = {
+export const kubernetesConfigSchema = {
   type: "object",
   properties: {
     version: { type: "string" },
-    serviceCidrRange: { type: "string", flag: "service-cidr-range" },
-    serviceCIDR: { type: "string", flag: "service-cidr" },
-    haCluster: { type: "boolean", flag: "ha" },
-    masterAddress: { type: "string", flag: "kuberenetes-master-address" },
-    loadBalancerAddress: { type: "string", flag: "load-balancer-address" },
-    bootstrapToken: { type: "string", flag: "bootstrap-token" },
-    bootstrapTokenTTL: { type: "string", flag: "bootstrap-token-ttl" },
-    kubeadmTokenCAHash: { type: "string", flag: "kubeadm-token-ca-hash" },
-    controlPlane: { type: "boolean", flag: "control-plane"},
-    certKey: { type: "string", flag: "cert-key" },
+    serviceCidrRange: { type: "string", flag: "service-cidr-range", description: "The size of the CIDR for Kubernetes (can be presented as just a number or with a preceding slash)" },
+    serviceCIDR: { type: "string", flag: "service-cidr", description: "This defines subnet for kubernetes" },
+    haCluster: { type: "boolean", flag: "ha", description: "Create the cluster as a high availability cluster (note that this needs a valid load balancer address and additional nodes to be a truly HA cluster)" },
+    masterAddress: { type: "string", flag: "kuberenetes-master-address", description: "The address of the internal Kubernetes API server, used during join scripts (read-only)" },
+    loadBalancerAddress: { type: "string", flag: "load-balancer-address", description: "Used for High Availability installs, indicates the address of the external load balancer" },
+    bootstrapToken: { type: "string", flag: "bootstrap-token", description: "A secret needed for new nodes to join an existing cluster (read-only)" },
```

Comment:
You can pass `bootstrap-token` to the install script on the first master. That will be used as the `kubeadm-token` that is passed to the join script for remote workers and masters. The `bootstrap-token` is usually generated by the install script but some people pre-generate it so they can automate bringing up a cluster. If you attempt to set the `kubeadm-token` flag on the join script and you hadn't passed the same value in the `bootstrap-token` flag to the install script the join will fail.

Reply:
All that to say it's not really `read-only`

Reply:
thanks that was really helpful, removing read-only tag